pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -430,7 +430,7 @@ func calculatePodRequests(pods []*v1.Pod, container string, resource v1.Resource
 				if containerRequest, ok := c.Resources.Requests[resource]; ok {
 					podSum += containerRequest.MilliValue()
 				} else {
-					return nil, fmt.Errorf("missing request for %s", resource)
+					return nil, fmt.Errorf("Container %s in POD %s missing request for %s", c.Name, pod.ObjectMeta.Name, resource)
```

Comment:
A couple of things:

1) rename POD to Pod, as it's officially spelled [doc](https://kubernetes.io/docs/concepts/workloads/pods/)
2) don't capitalize the error message ([reason](https://github.com/golang/go/wiki/CodeReviewComments#error-strings))
3) nit: I would reword into "missing request for ... in container ... of pod ..." so that the most important info (cause) comes first before container/pod context (where).

Reply:
Updated the diffs as per the comments.